src/optimizer/transforms/group-single-chars-to-char-class.js

```diff
@@ -8,49 +8,74 @@
 /**
  * A regexp-tree plugin to replace single char group disjunction to char group
  *
+ * a|b|c -> [abc]
+ * [12]|3|4 -> [1234]
  * (a|b|c) -> ([abc])
  * (?:a|b|c) -> [abc]
  */
 module.exports = {
-  Group(path) {
-    const {node} = path;
+  Disjunction(path) {
+    const {node, parent} = path;
+
+    if (!handlers[parent.type]) {
+      return;
+    }
 
     const charset = new Map();
 
     if (
-      !shouldProcessExpressionCharset(node.expression, charset) ||
+      !shouldProcess(node, charset) ||
       !charset.size
     ) {
       return;
     }
 
     const characterClass = {
       type: 'CharacterClass',
-      expressions: Array.from(charset.values())
+      expressions: Array.from(charset.keys())
+        .sort()
+        .map(key => charset.get(key))
     };
 
+    handlers[parent.type](path.getParent(), characterClass);
+  }
+};
+
+const handlers = {
+  RegExp: (path, characterClass) => {
```

Comment:
Let's use method notation on objects (to preserve `this` access; otherwise `this` is global in this case since captured lexically by arrow function)